### PR TITLE
Switch to PSR-4, null values in data providers

### PR DESCRIPTION
**Pull request by afurculita**
The current commit contains the following changes:
1. The switch to PSR-4 for spec/ and src/ folders
2. Added support for Symfony 3.0 in dev requirements
3. Added a description to this library in composer.json
4. Added support for "null" values provided by data providers
5. Many code improvements to DataProviderListener
6. Many code improvements to DataProviderMaintainer

### CODE REVIEW

**Comment by norberttech** [comment]
hi @afurculita from what I can see tests are failing, could you please loot at it?

**Comment by afurculita** [comment]
@norzechowicz it seems that the tests are not failing due to the changes I've made, but they are failing for the current master to. Maybe something changes in Phpspec. Can you please check? Maybe I'm missing something

**Comment by purificant** [comment]
@norzechowicz Tests are passing in master for me: https://travis-ci.org/purificant/phpspec-data-provider-extension/builds/142274892

File: composer.json [hunk 1]
```diff
@@ -1,8 +1,9 @@
 {
-    "name": "coduo/phpspec-data-provider-extension",
+    "name": "afurculita/phpspec-data-provider-extension",
```
Comment (norberttech) [inline composer.json:2]: ;> ?

File: composer.json [hunk 2]
```diff
@@ -14,22 +15,26 @@
...
-        "symfony/filesystem": "~2.3",
-        "symfony/process": "~2.3",
-        "symfony/console": "~2.3.2",
-        "bossa/phpspec2-expect": "^1"
+        "behat/behat": "~3.0",
+        "symfony/filesystem": "~2.3|~3.0",
+        "symfony/process": "~2.3|~3.0",
+        "symfony/console": "~2.3.2|~3.0",
+        "bossa/phpspec2-expect": "^1",
+        "phpspec/phpspec": "^2"
```
Comment (norberttech) [inline composer.json:26]: phpspec is required dependency, should be moved back to "require" section

File: composer.json [hunk 3]
```diff
@@ -14,22 +15,26 @@
...
-        "psr-0": { "Coduo\\PhpSpec": "src/"}
+        "psr-4": { "Coduo\\PhpSpec\\DataProvider\\": "src/"}
+    },
+    "autoload-dev": {
+        "psr-4": { "spec\\Coduo\\PhpSpec\\DataProvider\\": "spec/"}
     },
     "config": {
         "bin-dir": "bin"
     },
+    "prefer-stable": true,
```
Comment (norberttech) [inline composer.json:37]: I think composer prefer stable by default

File: .travis.yml [hunk 4]
```diff
@@ -4,14 +4,15 @@ php:
...
+  - 7.0
 
 env:
   - COMPOSER_OPTIONS='install --prefer-source'
 
 matrix:
    include:
      - php: 5.4
-       env: COMPOSER_OPTIONS='update --prefer-lowest --prefer-source'
+       env: COMPOSER_OPTIONS='update --prefer-source'
```
Comment (norberttech) [inline .travis.yml:15]: this should be reverted, prefer-lowest make us sure there are no BC breaks

**Pull request closed without merging**